### PR TITLE
2nd Assignment by 202403234 (Jihyun Yang)

### PR DESCRIPTION
**Pull request by angjihyun**
# Assignment Submission

## Student Information
- **Student ID**: [202403234]
- **Name**: [Jihyun Yang]
- **Assignment**: [Assignment2]

## Changed Files

```
[FILE_TREE_PLACEHOLDER]
```

### Problem 1
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 2
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 3
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 4 (Optional)
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 5 (Optional)
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 6 (Optional)
- [ ] Implemented required functionality
- [ ] All tests pass locally

## Testing
- [ ] I have run `python grader.py` locally
- [ ] All implemented problems pass their tests
- [ ] I have tested edge cases mentioned in the problem statements

## Additional Notes

---
**Note**: This PR will be automatically validated for:
- Correct directory structure
- Valid Python syntax
- PR title format: `Nth Assignment by {student_id} (Full Name)`
- File organization compliance

### CODE REVIEW

File: submissions/202403234/assignment2/problem3/main.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+Problem 3 — tokenstats (module + __main__ demo)
+- count token frequencies
+- get top-k (freq desc, token asc for ties)
+"""
+
+def count_tokens(tokens: list[str]) -> dict[str, int]:
+    # TODO: 구현하세요
+    # 힌트:
+    # 1) 빈 딕셔너리 생성: d = {}
+    d={} #d라는 딕셔너리 생섷함
```
Comment (hist0613) [inline submissions/202403234/assignment2/problem3/main.py:12]: ㅋㅋㅋ

File: submissions/202403234/assignment2/problem3/main.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    # 2) 각 토큰을 순회하면서 카운트: d[token] = d.get(token, 0) + 1
+    for token in tokens:
+        d[token]=d.get(token,0)+1
+    return d
+    # 3) 또는 collections.Counter 사용 가능 (하지만 직접 구현도 간단함)
```
Comment (hist0613) [inline submissions/202403234/assignment2/problem3/main.py:17]: 코드/주석 정리 부탁드립니당

**Comment by hist0613** [comment]
grader 기준 만점입니다.
코드 정리만 부탁드립니다.

**Review by okm0408 — COMMENTED** [review]
빈 문자열/공백 전용 문자열 처리 (if not s or s.strip() == "")와 단일 공백 분리 로직이 명확하게 구현되었습니다.
